src/Commands/AutodiscoveryPackageLockVerify.php

```diff
@@ -26,52 +25,54 @@ public function __construct(PackageManifest $manifest)
             $manifest->basePath,
             $manifest->manifestPath
         );
-        $this->warningMessages = new Collection();
+        $this->resolver = $resolver;
     }
 
     public function handle()
     {
         try {
-            $lockCollection = $this->packageManifest->collectManifestFromLock();
-            $autoloadCollection = $this->packageManifest->collectManifestFromComposerAutoload();
+            $errorMessages = $this->resolver->resolve($this->packageManifest);
 
-            $this->checkIfAutodiscoveredPackagesAreInLockfile($autoloadCollection, $lockCollection);
-            $this->checkIfLockedPackagesAreInAutoDiscovery($autoloadCollection, $lockCollection);
+            if (
+                $errorMessages->get(AutodiscoveryLockResolver::NOT_IN_AUTOLOAD)->isEmpty() &&
+                $errorMessages->get(AutodiscoveryLockResolver::NOT_IN_LOCK)->isEmpty()
+            ) {
+                $this->info('The lock file is up to date with the autodiscovered packages.');
 
-            if (count($this->warningMessages) > 0) {
-                $this->warn('The lock file is not up to date with the autodiscovered packages.');
-                $this->line('Run `php artisan autodiscovery:generate-lock` to update the lock file.');
-                $this->line('The following packages are not in sync:');
+                return self::SUCCESS;
+            }
+            $this->warn('The lock file is not up to date with the autodiscovered packages.');
 
-                $this->warningMessages->each(function ($message) {
-                    $this->warn($message);
-                });
+            $this->displayNotInAutoloadMessages($errorMessages->get(AutodiscoveryLockResolver::NOT_IN_AUTOLOAD));
+            $this->displayNotInLockMessages($errorMessages->get(AutodiscoveryLockResolver::NOT_IN_LOCK));
 
-                throw new \Exception('A missmatch between the lock file and the autodiscovered packages was found.');
-            } else {
-                $this->info('The lock file is up to date with the autodiscovered packages.');
-            }
-            return self::SUCCESS;
+            throw new \Exception('A mismatch between the lock file and the autodiscovered packages was found.');
```

Comment:
Generic exception is used for control flow, I think this should be moved outside of the `try` / `catch` and simply return `self::FAILURE`. Alternatively specific Exceptions could be used to provide better error messages

Reply:
I agree, it has currently felt a little bit clunky to me. I'll look into improving the flow!